### DISCUSSION TITLE
ADA Accessibility Compliance

### DISCUSSION THREAD

**Discussion by nickyketts** [elementor/elementor#18944] [Developers Community]
We are failing some Accessibility checks when completing an ADA Audit. 

1.	[aria-*] attributes do not match their roles - Each ARIA `role` supports a specific subset of `aria-*` attributes. Mismatching these invalidates the `aria-*` attributes. 

This refers to the Social Share buttons contained in the Share Buttons Element, for example the Facebook Share button:

<div class="elementor-share-btn elementor-share-btn_facebook" tabindex="0" aria-label="Share on facebook">
<span class="elementor-share-btn__icon">
<i class="fab fa-facebook" aria-hidden="true"></i>
</span>
</div>

2.	Buttons do not have an accessible name.

We have a search facility in the header of the website consisting of a text box and search button. It is not possible to add an accessible name to the search button. 

3.	Elements with an ARIA [role] that require children to contain a specific [role] are missing some or all of those required children.

This refers to the YouTube Playlist element, specifically the thumbnail for videos in the playlist

4.	ARIA IDs are not unique

Again refers to the YouTube playlist element. Same information is used for the video thumbnail and the video description:

div#e-tab-content-4591.e-tab-content.elementor-clearfix - thumbnail
div#e-tab-content-4591.e-tab-content.elementor-clearfix – description

5.	Image elements do not have [alt] attributes

Again refers to the thumbnail images in the YouTube playlist element.

6.	Heading elements are not in a sequentially-descending order.

Again refers to the YouTube playlist element. The playlist title is an H2 tag, the titles of the videos within the playlist are H4 tags.

Are there any recommended workarounds for these issues to ensure ADA Compliance, or can I request that these issues are looked at in future releases of Elementor.

**Comment by sarah-cul** [comment]
Another Accessibility Issue: Toggle Items that are created when using the Elementor plugin, cause the issue of having anchor elements that specify a tabindex but do not have an href. One of the solutions here is to add a role="link" attribute to the anchor element, but this is not possible from the elementor interface. Attempting to do this from Advanced custom attributes will put the attribute under a parent DIV, not the anchor.
